test/common/upstream/health_checker_impl_test.cc

```diff
@@ -1025,6 +1058,130 @@ TEST_F(HttpHealthCheckerImplTest, SuccessWithMultipleHostSets) {
   EXPECT_EQ(Host::Health::Healthy, cluster_->prioritySet().getMockHostSet(1)->hosts_[0]->health());
 }
 
+TEST_F(HttpHealthCheckerImplTest, SuccessExpectedResponseCheck) {
```

Comment:
Consider adding tests that send body in multiple small portions to validate that you concatenate it correctly in the health checker.

Reply:
Done.